.github/workflows/upload_description.yaml

```diff
@@ -0,0 +1,53 @@
+name: Upload description to Youtube
+
+on:
+  workflow_dispatch:
+  push:
+    #branches: [master]
+    paths:
+      - "blablas/**"
+
+jobs:
+  detect_changes:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: jitterbit/get-changed-files@v1
+        continue-on-error: true
+        id: files
+    outputs:
+      all: ${{ steps.files.outputs.all }}
+
+  upload_description:
+    runs-on: ubuntu-18.04
+    needs: detect_changes
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        name: Install node
+        with:
+          node-version: "14"
+          cache: "yarn"
+
+      - name: Install Deps
+        run: yarn install
+
+      - name: Extract descriptions and URLs
+        id: extract_descriptions_and_urls
+        env:
+          CLIENT_ID: ${{ secrets.CLIENT_ID }}
+          CLIENT_SECRET: ${{ secrets.CLIENT_SECRET }}
+          REDIRECT_URL: ${{ secrets.REDIRECT_URL }}
+          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+          REFRESH_TOKEN: ${{ secrets.REFRESH_TOKEN }}
+        run: |
+          echo "Changed files"
+          echo ${{ needs.detect_changes.outputs.all }}
+          for changed_file in ${{ needs.detect_changes.outputs.all }}; do
```

Comment:
I think using a script shell to get episode youtube URL is a little bit risky here, I would suggest using javascript for this may be using the same approach we did with the podcast script. 

here is my suggestion: 
```yml

- name: Extract descriptions and URLs
        id: extract_descriptions_and_urls
        env:
          CLIENT_ID: ${{ secrets.CLIENT_ID }}
          CLIENT_SECRET: ${{ secrets.CLIENT_SECRET }}
          REDIRECT_URL: ${{ secrets.REDIRECT_URL }}
          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
          REFRESH_TOKEN: ${{ secrets.REFRESH_TOKEN }}
          UPDATED_FILES: ${{ needs.detect_changes.outputs.all }}
        run:  yarn upload-description-to-youtube 
```

In this case, you will have all updated files as env variables and you can play with them in js script, then you can create a simple script to get episode details using `front-matter` module  like the following :  
```js
// get-episode-details.js

const getEpisodeDetails = (file) => {
  try {
    const fileContent = fs.readFileSync(file, "utf8")
    const description = markdownToTxt(fileContent)
    // read file meta
    const meta = fm(fileContent)
    const youtubeUrl = meta.attributes.youtube // you can get the id directly here
    const episode = { youtubeUrl, description }
    return episode
  } catch (error) {
    return null
  }
}


```

Reply:
Noice, TIL about front-matter.

I moved the shell logic to javascript in this commit : https://github.com/DevC-Casa/geeksblabla.com/pull/305/commits/7be875d203902593179d771da8cbe600d60d4526

But I didn't catch the exception inside it. Instead, I let it throw exceptions then handled them in the loop. Is that cool ?